Frontend/src/Components/Dashboard/Dashboard.js

```diff
@@ -95,70 +93,175 @@ export default function Dashboard(props) {
+          setDataView3(dataView2);
+        } else if (event.target.value === document.getElementById("dataViewSelect4").value) {
+          // If trying to switch to a data view that is already being displayed in the bottom right section, switch the
+          // data views in this section and the other one
+          setDataView4(dataView2);
+        }
       }
       setDataView2(event.target.value);
+    } else if (event.target.id === "dataViewSelect3") {
+      // Avoid duplicate data views, unless they are both empty
+      if(event.target.value !== "") {
+        if (event.target.value === document.getElementById("dataViewSelect1").value) {
+          // If trying to switch to a data view that is already being displayed in the top left section, switch the
+          // data views in this section and the other one
+          setDataView1(dataView3);
+        } else if (event.target.value === document.getElementById("dataViewSelect2").value) {
+          // If trying to switch to a data view that is already being displayed in the top right section, switch the
+          // data views in this section and the other one
+          setDataView2(dataView3);
+        } else if (event.target.value === document.getElementById("dataViewSelect4").value) {
+          // If trying to switch to a data view that is already being displayed in the bottom right section, switch the
+          // data views in this section and the other one
+          setDataView4(dataView3);
+        }
+      }
+      setDataView3(event.target.value);
+    } else if (event.target.id === "dataViewSelect4") {
+      // Avoid duplicate data views, unless they are both empty
+      if(event.target.value !== "") {
+        if (event.target.value === document.getElementById("dataViewSelect1").value) {
+          // If trying to switch to a data view that is already being displayed in the top left section, switch the
+          // data views in this section and the other one
+          setDataView1(dataView4);
+        } else if (event.target.value === document.getElementById("dataViewSelect2").value) {
+          // If trying to switch to a data view that is already being displayed in the top right section, switch the
+          // data views in this section and the other one
+          setDataView2(dataView4);
+        } else if (event.target.value === document.getElementById("dataViewSelect3").value) {
+          // If trying to switch to a data view that is already being displayed in the bottom left section, switch the
+          // data views in this section and the other one
+          setDataView3(dataView4);
+        }
+      }
+      setDataView4(event.target.value);
     }
   };
 
   // Choose the data view to return/display based on the given option
   const switchDataView = (optionValue) => {
-    if (optionValue === "general") {
-      return <DataView data={state.data} />;
-    } else if (optionValue === "battery") {
+    if (optionValue === "battery_pack") {

```

Comment:
We should use some sort of enum here - that way if we change the option value on where it's being set - it'll also change what's expected.